.github/workflows/release.yml

```diff
@@ -1,17 +1,43 @@
 name: Publish Package to npmjs
 on:
-  release:
-    types: [published]
+  push:
+    tags:
+      - "v*"
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      # Setup .npmrc file to publish to npm
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - name: set node-version
+        uses: actions/setup-node@v4
+        with:
+          registry-url: 'https://registry.npmjs.org/'
+          node-version-file: 'package.json'
+      - name: Build
+        run: make build
+      - name: Prettier
+        run: make prettier
```

Comment:
this shouldnt be in the publish workflow. we should be publishing the same code we test. Im in favor of adding this to a pre-merge workflow and failing if any changes are detected.